In [3]:
import matplotlib.pyplot as plt
%matplotlib inline
import pandas as pd
import numpy as np
import seaborn as sns
import sqlalchemy as sa

%config IPCompleter.greedy=True

pd.options.display.float_format='{:,.2f}'.format

# 1. Carga de datos

In [4]:
con=sa.create_engine('sqlite:///../cas3/ecommerce.db')

In [5]:
from sqlalchemy import inspect
ins=inspect(con)
ins.get_table_names()

['2019-Dec', '2019-Nov', '2019-Oct', '2020-Feb', '2020-Jan']

In [6]:
oct=pd.read_sql('2019-Oct',con)
nov=pd.read_sql('2019-Nov',con)
dic=pd.read_sql('2019-Dec',con)
ene=pd.read_sql('2020-Jan',con)
feb=pd.read_sql('2020-Feb',con)

# 2. Integración datos

In [7]:
oct.columns

Index(['index', 'event_time', 'event_type', 'product_id', 'category_id',
       'category_code', 'brand', 'price', 'user_id', 'user_session'],
      dtype='object')

In [8]:
nov.columns

Index(['index', 'event_time', 'event_type', 'product_id', 'category_id',
       'category_code', 'brand', 'price', 'user_id', 'user_session'],
      dtype='object')

In [10]:
dic.columns

Index(['index', 'event_time', 'event_type', 'product_id', 'category_id',
       'category_code', 'brand', 'price', 'user_id', 'user_session'],
      dtype='object')

In [11]:
ene.columns

Index(['index', 'event_time', 'event_type', 'product_id', 'category_id',
       'category_code', 'brand', 'price', 'user_id', 'user_session'],
      dtype='object')

In [12]:
feb.columns

Index(['index', 'event_time', 'event_type', 'product_id', 'category_id',
       'category_code', 'brand', 'price', 'user_id', 'user_session'],
      dtype='object')

COmparten todos las mismas columnas por lo que podemos apilar los diferentes dataframes.

In [17]:
df=pd.concat([oct,nov,dic,ene,feb],axis=0)
df

index               event_time event_type  product_id  \
0            68  2019-10-01 00:01:46 UTC       view     5843665   
1            72  2019-10-01 00:01:55 UTC       cart     5868461   
2            95  2019-10-01 00:02:50 UTC       view     5877456   
3           122  2019-10-01 00:03:41 UTC       view     5649270   
4           124  2019-10-01 00:03:44 UTC       view       18082   
...         ...                      ...        ...         ...   
429785  4156660  2020-02-29 23:58:49 UTC       cart     5815662   
429786  4156663  2020-02-29 23:58:57 UTC       view     5815665   
429787  4156668  2020-02-29 23:59:05 UTC       cart     5815665   
429788  4156675  2020-02-29 23:59:28 UTC       view     5817692   
429789  4156680  2020-02-29 23:59:54 UTC       view     5716351   

                category_id category_code     brand  price    user_id  \
0       1487580005092295511          None     f.o.x   9.44  462033176   
1       1487580013069861041          None   italwax   3.57  514753614   
2       1487580006300255120          None  jessnail 122.22  527418424   
3       1487580013749338323          None   concept   6.19  555448072   
4       1487580005411062629          None       cnd  16.03  552006247   
...                     ...           ...       ...    ...        ...   
429785  1487580006317032337          None      None   0.92  147995998   
429786  1487580006317032337          None      None   0.59  147995998   
429787  1487580006317032337          None      None   0.59  147995998   
429788  1487580010872045658          None      None   0.79  619841242   
429789  1487580010872045658          None     irisk   0.79  619841242   

                                user_session  
0       a18e0999-61a1-4218-8f8f-61ec1d375361  
1       e2fecb2d-22d0-df2c-c661-15da44b3ccf1  
2       86e77869-afbc-4dff-9aa2-6b7dd8c90770  
3       b5f72ceb-0730-44de-a932-d16db62390df  
4       2d8f304b-de45-4e59-8f40-50c603843fe5  
...                                      ...  
429785  5ff96629-3627-493e-a25b-5a871ec78c90  
429786  5ff96629-3627-493e-a25b-5a871ec78c90  
429787  5ff96629-3627-493e-a25b-5a871ec78c90  
429788  18af673b-7fb9-4202-a66d-5c855bc0fd2d  
429789  18af673b-7fb9-4202-a66d-5c855bc0fd2d  

[2095076 rows x 10 columns]

# 3. Calidad de datos

## 3.1 Cambio tipo variables

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2095076 entries, 0 to 429789
Data columns (total 10 columns):
 #   Column         Dtype  
---  ------         -----  
 0   index          int64  
 1   event_time     object 
 2   event_type     object 
 3   product_id     int64  
 4   category_id    int64  
 5   category_code  object 
 6   brand          object 
 7   price          float64
 8   user_id        int64  
 9   user_session   object 
dtypes: float64(1), int64(4), object(5)
memory usage: 175.8+ MB


In [19]:
#Elimino columna index ya que no aporta info y pondré event_time con index
df.drop(columns='index',inplace=True)

In [20]:
df['event_time']=pd.to_datetime(df.event_time)

## 3.2 Cambio nombre variables

In [21]:
df.columns = ['fecha',
              'evento',
              'producto',
              'categoria',
              'categoria_cod',
              'marca',
              'precio',
              'usuario',
              'sesion']
df

fecha evento  producto            categoria  \
0      2019-10-01 00:01:46+00:00   view   5843665  1487580005092295511   
1      2019-10-01 00:01:55+00:00   cart   5868461  1487580013069861041   
2      2019-10-01 00:02:50+00:00   view   5877456  1487580006300255120   
3      2019-10-01 00:03:41+00:00   view   5649270  1487580013749338323   
4      2019-10-01 00:03:44+00:00   view     18082  1487580005411062629   
...                          ...    ...       ...                  ...   
429785 2020-02-29 23:58:49+00:00   cart   5815662  1487580006317032337   
429786 2020-02-29 23:58:57+00:00   view   5815665  1487580006317032337   
429787 2020-02-29 23:59:05+00:00   cart   5815665  1487580006317032337   
429788 2020-02-29 23:59:28+00:00   view   5817692  1487580010872045658   
429789 2020-02-29 23:59:54+00:00   view   5716351  1487580010872045658   

       categoria_cod     marca  precio    usuario  \
0               None     f.o.x    9.44  462033176   
1               None   italwax    3.57  514753614   
2               None  jessnail  122.22  527418424   
3               None   concept    6.19  555448072   
4               None       cnd   16.03  552006247   
...              ...       ...     ...        ...   
429785          None      None    0.92  147995998   
429786          None      None    0.59  147995998   
429787          None      None    0.59  147995998   
429788          None      None    0.79  619841242   
429789          None     irisk    0.79  619841242   

                                      sesion  
0       a18e0999-61a1-4218-8f8f-61ec1d375361  
1       e2fecb2d-22d0-df2c-c661-15da44b3ccf1  
2       86e77869-afbc-4dff-9aa2-6b7dd8c90770  
3       b5f72ceb-0730-44de-a932-d16db62390df  
4       2d8f304b-de45-4e59-8f40-50c603843fe5  
...                                      ...  
429785  5ff96629-3627-493e-a25b-5a871ec78c90  
429786  5ff96629-3627-493e-a25b-5a871ec78c90  
429787  5ff96629-3627-493e-a25b-5a871ec78c90  
429788  18af673b-7fb9-4202-a66d-5c855bc0fd2d  
429789  18af673b-7fb9-4202-a66d-5c855bc0fd2d  

[2095076 rows x 9 columns]

## 3.3 Analisi nulos

In [22]:
df.isna().sum()

fecha                  0
evento                 0
producto               0
categoria              0
categoria_cod    2060411
marca             891646
precio                 0
usuario                0
sesion               506
dtype: int64

Conclusiones:

* categoria_cod tiene casi todos los registros a nulo
* marca tiene casi la mitad de los registros a nulo
* hay 500 nulos en sesión

Acciones:

* eliminar las variables categoria_cod y marca
* eliminar los nulos de sesión ya que es una variable relevante

In [23]:
df=df.drop(columns=['categoria_cod','marca']).dropna()
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2094570 entries, 0 to 429789
Data columns (total 7 columns):
 #   Column     Dtype              
---  ------     -----              
 0   fecha      datetime64[ns, UTC]
 1   evento     object             
 2   producto   int64              
 3   categoria  int64              
 4   precio     float64            
 5   usuario    int64              
 6   sesion     object             
dtypes: datetime64[ns, UTC](1), float64(1), int64(3), object(2)
memory usage: 127.8+ MB


In [24]:
df.isna().sum()

fecha        0
evento       0
producto     0
categoria    0
precio       0
usuario      0
sesion       0
dtype: int64

## 3.4 Analisi variables numericas

In [25]:
df.describe().T

count                         mean  \
producto  2,094,570.00                 5,487,103.56   
categoria 2,094,570.00 1,553,112,489,392,098,048.00   
precio    2,094,570.00                         8.42   
usuario   2,094,570.00               521,077,545.56   

                                 std                          min  \
producto                1,300,923.90                     3,752.00   
categoria 167,907,497,920,480,576.00 1,487,580,004,807,082,752.00   
precio                         19.14                       -47.62   
usuario                87,553,855.76                 4,661,182.00   

                                   25%                          50%  \
producto                  5,724,652.00                 5,811,665.00   
categoria 1,487,580,005,754,995,456.00 1,487,580,008,246,412,288.00   
precio                            2.05                         4.00   
usuario                 480,613,387.00               553,341,613.00   

                                   75%                          max  
producto                  5,858,353.00                 5,932,595.00  
categoria 1,487,580,013,489,291,520.00 2,242,903,426,784,559,104.00  
precio                            6.86                       327.78  
usuario                 578,406,571.00               622,087,993.00

La unica variable posible de analizar es el precio. 

Veo precios negativos. No tiene mucho sentido, voy a analizarlo a continuación.

In [26]:
df[df.precio<=0]

fecha evento  producto            categoria  \
343    2019-10-01 02:15:41+00:00   view   5892052  1487580010377117763   
924    2019-10-01 05:16:30+00:00   view   5889621  1487580010561667147   
933    2019-10-01 05:18:03+00:00   view   5889622  1487580010561667147   
937    2019-10-01 05:18:46+00:00   view   5889623  1487580010561667147   
1077   2019-10-01 05:38:01+00:00   view   5889627  1487580010561667147   
...                          ...    ...       ...                  ...   
428011 2020-02-29 20:04:49+00:00   cart   5824841  1897124478404526487   
428012 2020-02-29 20:04:49+00:00   cart   5826413  1487580005511725929   
428013 2020-02-29 20:04:49+00:00   cart   5832437  1487580007675986893   
428014 2020-02-29 20:04:49+00:00   cart   5851606  2055161088059638328   
428370 2020-02-29 20:26:16+00:00   view   5923106  1487580008246412266   

        precio    usuario                                sesion  
343       0.00  555455025  320f6021-30ac-4a58-ae17-bac1cc32aac3  
924       0.00  523988665  00849bd2-fcd2-4cb4-af31-4e264f151848  
933       0.00  523988665  80cfe614-f0a5-4101-a2b6-a21227590470  
937       0.00  523988665  c2cd0464-3d2b-48e2-9667-bac248fe297a  
1077      0.00  523988665  8b2bf9d8-43f0-43b2-bed3-13b2c956cada  
...        ...        ...                                   ...  
428011    0.00  469761446  8bf369b4-92c0-4fb8-88a5-8a2dd0947e46  
428012    0.00  469761446  8bf369b4-92c0-4fb8-88a5-8a2dd0947e46  
428013    0.00  469761446  8bf369b4-92c0-4fb8-88a5-8a2dd0947e46  
428014    0.00  469761446  8bf369b4-92c0-4fb8-88a5-8a2dd0947e46  
428370    0.00  622047714  74f04dc6-2b3c-4565-beda-f575d73ed81c  

[20544 rows x 7 columns]

Son 20544 registros de 2 millones. EL 1%, lo podria eliminar.

Voy a analizar si se trata de algun producto en concreto.

In [27]:
df[df.precio<=0].producto.value_counts()

producto
5896186    79
5903915    50
5873428    37
5851294    29
5851304    29
           ..
5859332     1
5859329     1
5858607     1
5848998     1
5832437     1
Name: count, Length: 7049, dtype: int64

No parece que sea problema de un producto concreto, así que vamos a eliminar todos los registros.

In [28]:
df = df[df.precio > 0]
df

fecha evento  producto            categoria  \
0      2019-10-01 00:01:46+00:00   view   5843665  1487580005092295511   
1      2019-10-01 00:01:55+00:00   cart   5868461  1487580013069861041   
2      2019-10-01 00:02:50+00:00   view   5877456  1487580006300255120   
3      2019-10-01 00:03:41+00:00   view   5649270  1487580013749338323   
4      2019-10-01 00:03:44+00:00   view     18082  1487580005411062629   
...                          ...    ...       ...                  ...   
429785 2020-02-29 23:58:49+00:00   cart   5815662  1487580006317032337   
429786 2020-02-29 23:58:57+00:00   view   5815665  1487580006317032337   
429787 2020-02-29 23:59:05+00:00   cart   5815665  1487580006317032337   
429788 2020-02-29 23:59:28+00:00   view   5817692  1487580010872045658   
429789 2020-02-29 23:59:54+00:00   view   5716351  1487580010872045658   

        precio    usuario                                sesion  
0         9.44  462033176  a18e0999-61a1-4218-8f8f-61ec1d375361  
1         3.57  514753614  e2fecb2d-22d0-df2c-c661-15da44b3ccf1  
2       122.22  527418424  86e77869-afbc-4dff-9aa2-6b7dd8c90770  
3         6.19  555448072  b5f72ceb-0730-44de-a932-d16db62390df  
4        16.03  552006247  2d8f304b-de45-4e59-8f40-50c603843fe5  
...        ...        ...                                   ...  
429785    0.92  147995998  5ff96629-3627-493e-a25b-5a871ec78c90  
429786    0.59  147995998  5ff96629-3627-493e-a25b-5a871ec78c90  
429787    0.59  147995998  5ff96629-3627-493e-a25b-5a871ec78c90  
429788    0.79  619841242  18af673b-7fb9-4202-a66d-5c855bc0fd2d  
429789    0.79  619841242  18af673b-7fb9-4202-a66d-5c855bc0fd2d  

[2074026 rows x 7 columns]

## 3.5 Analisi variables categoricas

In [29]:
df.select_dtypes(include=['object','category'])

evento                                sesion
0        view  a18e0999-61a1-4218-8f8f-61ec1d375361
1        cart  e2fecb2d-22d0-df2c-c661-15da44b3ccf1
2        view  86e77869-afbc-4dff-9aa2-6b7dd8c90770
3        view  b5f72ceb-0730-44de-a932-d16db62390df
4        view  2d8f304b-de45-4e59-8f40-50c603843fe5
...       ...                                   ...
429785   cart  5ff96629-3627-493e-a25b-5a871ec78c90
429786   view  5ff96629-3627-493e-a25b-5a871ec78c90
429787   cart  5ff96629-3627-493e-a25b-5a871ec78c90
429788   view  18af673b-7fb9-4202-a66d-5c855bc0fd2d
429789   view  18af673b-7fb9-4202-a66d-5c855bc0fd2d

[2074026 rows x 2 columns]

In [30]:
df.evento.value_counts()

evento
view                961558
cart                574547
remove_from_cart    410357
purchase            127564
Name: count, dtype: int64

## 3.6 Set index

In [31]:
df.set_index('fecha', inplace = True)
df

evento  producto            categoria  precio  \
fecha                                                                     
2019-10-01 00:01:46+00:00   view   5843665  1487580005092295511    9.44   
2019-10-01 00:01:55+00:00   cart   5868461  1487580013069861041    3.57   
2019-10-01 00:02:50+00:00   view   5877456  1487580006300255120  122.22   
2019-10-01 00:03:41+00:00   view   5649270  1487580013749338323    6.19   
2019-10-01 00:03:44+00:00   view     18082  1487580005411062629   16.03   
...                          ...       ...                  ...     ...   
2020-02-29 23:58:49+00:00   cart   5815662  1487580006317032337    0.92   
2020-02-29 23:58:57+00:00   view   5815665  1487580006317032337    0.59   
2020-02-29 23:59:05+00:00   cart   5815665  1487580006317032337    0.59   
2020-02-29 23:59:28+00:00   view   5817692  1487580010872045658    0.79   
2020-02-29 23:59:54+00:00   view   5716351  1487580010872045658    0.79   

                             usuario                                sesion  
fecha                                                                       
2019-10-01 00:01:46+00:00  462033176  a18e0999-61a1-4218-8f8f-61ec1d375361  
2019-10-01 00:01:55+00:00  514753614  e2fecb2d-22d0-df2c-c661-15da44b3ccf1  
2019-10-01 00:02:50+00:00  527418424  86e77869-afbc-4dff-9aa2-6b7dd8c90770  
2019-10-01 00:03:41+00:00  555448072  b5f72ceb-0730-44de-a932-d16db62390df  
2019-10-01 00:03:44+00:00  552006247  2d8f304b-de45-4e59-8f40-50c603843fe5  
...                              ...                                   ...  
2020-02-29 23:58:49+00:00  147995998  5ff96629-3627-493e-a25b-5a871ec78c90  
2020-02-29 23:58:57+00:00  147995998  5ff96629-3627-493e-a25b-5a871ec78c90  
2020-02-29 23:59:05+00:00  147995998  5ff96629-3627-493e-a25b-5a871ec78c90  
2020-02-29 23:59:28+00:00  619841242  18af673b-7fb9-4202-a66d-5c855bc0fd2d  
2020-02-29 23:59:54+00:00  619841242  18af673b-7fb9-4202-a66d-5c855bc0fd2d  

[2074026 rows x 6 columns]

# 4.TRANSFORMACION DE DATOS

Vamos a crear 3 tipos de nuevas variables

* Extraer componentes
* Variables de calendario: Festivos locales (Rusia, ya que el ecommerce es ruso)
* Indicadores exógenos: Días no necesariamente festivos pero con interés comercial: Black Friday, Cyber Monday, Reyes, San Valentin

In [32]:
def comp_fecha(dataf):
    date = dataf.index.date
    año = dataf.index.year
    mes = dataf.index.month
    dia = dataf.index.day
    hora = dataf.index.hour
    minuto = dataf.index.minute
    segundo = dataf.index.second

    return(pd.DataFrame({'date':date, 'año':año,'mes':mes, 'dia':dia, 'hora':hora, 'minuto':minuto, 'segundo':segundo}))

In [74]:
df2=pd.concat([df.reset_index(),comp_fecha(df)],axis=1).set_index('fecha')
df2.isna().sum()

evento          0
producto        0
categoria       0
precio          0
usuario         0
sesion          0
black_friday    0
san_valentin    0
date            0
año             0
mes             0
dia             0
hora            0
minuto          0
segundo         0
dtype: int64

In [36]:
import holidays
festivo_ru = holidays.RU(years=2020)
festivo_ru

{datetime.date(2020, 1, 1): 'Новогодние каникулы', datetime.date(2020, 1, 2): 'Новогодние каникулы', datetime.date(2020, 1, 3): 'Новогодние каникулы', datetime.date(2020, 1, 4): 'Новогодние каникулы', datetime.date(2020, 1, 5): 'Новогодние каникулы', datetime.date(2020, 1, 6): 'Новогодние каникулы', datetime.date(2020, 1, 8): 'Новогодние каникулы', datetime.date(2020, 1, 7): 'Рождество Христово', datetime.date(2020, 2, 23): 'День защитника Отечества', datetime.date(2020, 3, 8): 'Международный женский день', datetime.date(2020, 5, 1): 'Праздник Весны и Труда', datetime.date(2020, 5, 9): 'День Победы', datetime.date(2020, 6, 12): 'День России', datetime.date(2020, 11, 4): 'День народного единства'}

In [75]:
df2['festivo_ru']=df2.date.apply(lambda x:1 if (x in festivo_ru) else 0)
df2.festivo_ru.value_counts()

festivo_ru
0    1953557
1     120469
Name: count, dtype: int64

In [77]:
df2['festivo']=np.where(df2.date.isin(festivo_ru),1,0)
df2.festivo.value_counts()

festivo
0    1953557
1     120469
Name: count, dtype: int64

In [78]:
#Comprobamos festivos
df2[df2.festivo == 1].date.value_counts().sort_index()

date
2019-11-04    16430
2020-01-01     7644
2020-01-02    10776
2020-01-03    10617
2020-01-04    13084
2020-01-05    14554
2020-01-06    10621
2020-01-07    12922
2020-01-08    14004
2020-02-23     9817
Name: count, dtype: int64

In [79]:
#Añado black friday y san valentin
df2['black_friday'] = 0
df2.loc['2019-11-29','black_friday'] = 1

df2['san_valentin'] = 0
df2.loc['2020-02-14','san_valentin'] = 1


In [80]:
df2.black_friday.value_counts()

black_friday
0    2051695
1      22331
Name: count, dtype: int64

# 5. Retoques finales

In [82]:
variables = df2.columns.to_list()
variables

['evento',
 'producto',
 'categoria',
 'precio',
 'usuario',
 'sesion',
 'black_friday',
 'san_valentin',
 'date',
 'año',
 'mes',
 'dia',
 'hora',
 'minuto',
 'segundo',
 'festivo_ru',
 'festivo']

In [83]:
orden = ['usuario',
         'sesion',
         'categoria',
         'evento',
         'producto',
         'precio']

orden

['usuario', 'sesion', 'categoria', 'evento', 'producto', 'precio']

In [84]:
resto = [nombre for nombre in variables if nombre not in orden]

resto

['black_friday',
 'san_valentin',
 'date',
 'año',
 'mes',
 'dia',
 'hora',
 'minuto',
 'segundo',
 'festivo_ru',
 'festivo']

In [85]:
df2 = df2[orden + resto]
df2

usuario                                sesion  \
fecha                                                                        
2019-10-01 00:01:46+00:00  462033176  a18e0999-61a1-4218-8f8f-61ec1d375361   
2019-10-01 00:01:55+00:00  514753614  e2fecb2d-22d0-df2c-c661-15da44b3ccf1   
2019-10-01 00:02:50+00:00  527418424  86e77869-afbc-4dff-9aa2-6b7dd8c90770   
2019-10-01 00:03:41+00:00  555448072  b5f72ceb-0730-44de-a932-d16db62390df   
2019-10-01 00:03:44+00:00  552006247  2d8f304b-de45-4e59-8f40-50c603843fe5   
...                              ...                                   ...   
2020-02-29 23:58:49+00:00  147995998  5ff96629-3627-493e-a25b-5a871ec78c90   
2020-02-29 23:58:57+00:00  147995998  5ff96629-3627-493e-a25b-5a871ec78c90   
2020-02-29 23:59:05+00:00  147995998  5ff96629-3627-493e-a25b-5a871ec78c90   
2020-02-29 23:59:28+00:00  619841242  18af673b-7fb9-4202-a66d-5c855bc0fd2d   
2020-02-29 23:59:54+00:00  619841242  18af673b-7fb9-4202-a66d-5c855bc0fd2d   

                                     categoria evento  producto  precio  \
fecha                                                                     
2019-10-01 00:01:46+00:00  1487580005092295511   view   5843665    9.44   
2019-10-01 00:01:55+00:00  1487580013069861041   cart   5868461    3.57   
2019-10-01 00:02:50+00:00  1487580006300255120   view   5877456  122.22   
2019-10-01 00:03:41+00:00  1487580013749338323   view   5649270    6.19   
2019-10-01 00:03:44+00:00  1487580005411062629   view     18082   16.03   
...                                        ...    ...       ...     ...   
2020-02-29 23:58:49+00:00  1487580006317032337   cart   5815662    0.92   
2020-02-29 23:58:57+00:00  1487580006317032337   view   5815665    0.59   
2020-02-29 23:59:05+00:00  1487580006317032337   cart   5815665    0.59   
2020-02-29 23:59:28+00:00  1487580010872045658   view   5817692    0.79   
2020-02-29 23:59:54+00:00  1487580010872045658   view   5716351    0.79   

                           black_friday  san_valentin        date   año  mes  \
fecha                                                                          
2019-10-01 00:01:46+00:00             0             0  2019-10-01  2019   10   
2019-10-01 00:01:55+00:00             0             0  2019-10-01  2019   10   
2019-10-01 00:02:50+00:00             0             0  2019-10-01  2019   10   
2019-10-01 00:03:41+00:00             0             0  2019-10-01  2019   10   
2019-10-01 00:03:44+00:00             0             0  2019-10-01  2019   10   
...                                 ...           ...         ...   ...  ...   
2020-02-29 23:58:49+00:00             0             0  2020-02-29  2020    2   
2020-02-29 23:58:57+00:00             0             0  2020-02-29  2020    2   
2020-02-29 23:59:05+00:00             0             0  2020-02-29  2020    2   
2020-02-29 23:59:28+00:00             0             0  2020-02-29  2020    2   
2020-02-29 23:59:54+00:00             0             0  2020-02-29  2020    2   

                           dia  hora  minuto  segundo  festivo_ru  festivo  
fecha                                                                       
2019-10-01 00:01:46+00:00    1     0       1       46           0        0  
2019-10-01 00:01:55+00:00    1     0       1       55           0        0  
2019-10-01 00:02:50+00:00    1     0       2       50           0        0  
2019-10-01 00:03:41+00:00    1     0       3       41           0        0  
2019-10-01 00:03:44+00:00    1     0       3       44           0        0  
...                        ...   ...     ...      ...         ...      ...  
2020-02-29 23:58:49+00:00   29    23      58       49           0        0  
2020-02-29 23:58:57+00:00   29    23      58       57           0        0  
2020-02-29 23:59:05+00:00   29    23      59        5           0        0  
2020-02-29 23:59:28+00:00   29    23      59       28           0        0  
2020-02-29 23:59:54+00:00   29    23      59       54           0        

In [87]:
df2.to_pickle('../cas3/df.pickle')